typescript-json-schema.ts

```diff
@@ -7,11 +7,13 @@ import { JSONSchema7 } from "json-schema";
 export { Program, CompilerOptions, Symbol } from "typescript";
 
 const vm = require("vm");
+require("ts-node/register");
```

Comment:
I don't think this should be here.

Reply:
This part is really necessary because of require typescript files. Without this part isn't ts-node registered and can't resolve typescript files.

Reply:
I don't think that's true. It worked before, right?

Reply:
Yes, but not when you try to require a typescript file. You could require only JS file. An in this case people will want to require (or import) file from they own project, such an example from its own repo. And in that case without ts-node won't work. You can try by yourself, remove this part and run tests. You will get an error that module can't be found.

Reply:
You can either import the js files we compile and put on npm or call the ts file with ts node. This is how it always worked and I don't want to change it.

Reply:
If it's needed for the require calls, you should figure out a way to only activate ts node for those calls.

Reply:
Aha ok, I think we both are looking on this problem different way. You suppose that this package will be used somewhere live, maybe on some web page? But I assume and for our purpose is need to use the "typescript-json-schema" only internally for generating json schemas before deploy. So from our side we don't have compiled scripts when is json generation called. Therefore I used this possibility to load typescript files. But I founded that in this case we need to require "ts-node/register" before using 'generateSchema'. So ok, I removed this part and for people who want to use this functionality internally, must use require("ts-node/register") before using "typescript-json-schema". I updated tests and put the test at the end of tests.

Reply:
I don't think that's an option as I explained below. Maybe we need to add this back but you also need to make ts-node a dependency then (which I find a bit worrisome since it's a complex dependency).

Reply:
It's true, so I added tsNodeRegister option to enable ts-node/register. This will works with command such as "typescript-json-schema src/generated-type.ts ShapesData --tsNodeRegister". Or it's possible to set it in settings when calling 'generateSchema()'. Maybe this will be better option then making ts-node as dependency, because in production, nobody will have ts scripts.

Reply:
Good idea.

Reply:
You have absolutely right, I make ts-node as dependency and now it's working correctly. Unfortunately I haven't found any better way.